### PR TITLE
adding authorization

### PR DESCRIPTION
**Pull request by kcalva**
checks for more security ,seeing UI pages based on user (and whether you can see other's stuff based on public or private account)

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:destroy, :create]
```
Comment (rayguna) [inline app/controllers/comments_controller.rb:3]: A neat use of before_action filter.

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
-
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (rayguna) [inline app/controllers/photos_controller.rb:3]: Another neat use of before_action filter.

File: config/routes.rb
Comment (rayguna) [inline config/routes.rb:1]: A neat way of incorporating except and only to limit the routes.

**Review by rayguna — APPROVED** [review]

**Review by ahuynh3a — APPROVED** [review]
Great job!

**Review by CarlosCastanedaDev — APPROVED** [review]
Great job, Kevin!

**Pull request merged**